smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -63,10 +43,13 @@
 /**
  * Abstract implementation useful for all protocols that use HTTP bindings.
  */
-public abstract class HttpBindingProtocolGenerator implements ProtocolGenerator {
+public abstract class HttpBindingProtocolGenerator implements ProtocolGenerator, HttpValueProvider {
```

Comment:
There's a *lot* of protected methods on this class. I'm wondering if there's a way to trim this number down using composition instead.